ipaserver/install/cainstance.py

```diff
@@ -1273,6 +1277,31 @@ def __setup_replication(self):
         # Activate Topology for o=ipaca segments
         self.__update_topology()
 
+    def __replica_ignore_initial_time_skew(self):
+        """
+        Set nsslapd-ignore-time-skew = on if not already set
+        and store the initial value in order to restore it later.
+
+        The on value allows replica initialization even if there
+        are excessive time skews.
+        """
+        dn = DN(('cn', 'config'))
+        entry_attrs = api.Backend.ldap2.get_entry(dn)
+        self.original_time_skew = entry_attrs['nsslapd-ignore-time-skew'][0]
+        if self.original_time_skew != 'on':
+            entry_attrs['nsslapd-ignore-time-skew'] = 'on'
+            api.Backend.ldap2.update_entry(entry_attrs)
+
+    def __replica_revert_time_skew(self):
+        """
+        Revert nsslapd-ignore-time-skew to its previous value.
+        """
+        dn = DN(('cn', 'config'))
+        entry_attrs = api.Backend.ldap2.get_entry(dn)
+        if self.original_time_skew != 'on':
+            entry_attrs['nsslapd-ignore-time-skew'] = self.original_time_skew
+            api.Backend.ldap2.update_entry(entry_attrs)
+
```

Comment:
The change looks fine and is perhaps superior to the version in dsinstance in that all the code is together and doesn't require a context switch to find the ldif.
I like that this is smart enough to not enable if enabled and not disable if not enabled. I assume a user could disable this setting via the --dirsrv-config option and by checking we don't raise an exception.
I wonder if one version should be added to ipaserver/install/service.py::Service so the code can live in one place so we'd only have the steps which would call the class version.

Reply:
Thanks for the suggestion. I put the code in the Service class and changed dsinstance so that it calls the methods from Service instead of writing its own methods.

Reply:
The bonus is that another issue is fixed with this patch: [RHEL-4879](https://issues.redhat.com/browse/RHEL-4879) RFE - Keep the configured value for the "nsslapd-ignore-time-skew" after a "force-sync".